course4/homework1/vector2d.h

```diff
@@ -2,8 +2,29 @@
 
 #include <ostream>
 
+
 struct Vector2d
 {
-	// TODO
+  int X;
+  int Y;
+  Vector2d operator+(const Vector2d& a)
+  {
+    Vector2d r;
+    r.X = X + a.X;
+    r.Y = Y + a.Y;
+    return r;
+  }
+  
+  bool operator==(const Vector2d& a)
+  {
+    return X == a.X and Y == a.Y;
```

Comment:
Don't use the textual form of 'and', use '&&' instead. Some compilers get stupid when they see 'and', and some libraries define macro/function with that name.